learn/shorts/intro-gh-actions.qmd

```diff
@@ -0,0 +1,369 @@
+---
+title: "An brief introduction to automation using GitHub Actions"
+description: |
+  GitHub Actions are a powerful way of running code or a series of tasks on an 
+  independent and fresh computing environment. You can use these as a way of 
+  automating tasks, running tasks on a schedule, testing your code, or any number
+  of things that you'd rather a computer do that you do manually.
+author: "Luke W. Johnston"
+date: "2024-05-21"
+date-modified: last-modified
+categories: 
+  - github actions
+  - automation
+  - servers
+  - workflows
+---
+
+## Motivation for this micro-lesson
+
+-   In our team, and in programming or software development in general,
+    we often need to frequently run code every time a specific event
+    happens (like someone makes changes to code) or we want to automate
+    a task that needs to happen on a regular basis. To do that, we need
+    to use tools that make it easier to automate those tasks. This
+    lesson aims to share our knowledge on using one of these tools
+    called [GitHub Actions](https://github.com/features/actions).
+
+## Assumed knowledge of reader
+
+-   No knowledge of any specific software or programming language is
+    needed, including Git.
+-   A general understanding of what "automation" means and answering
+    questions like:
+    -   How might a computer (conceptually) do a task for you, instead
+        of you doing it manually?
+    -   How could you (conceptually) get a computer to do a task for you
+        on a schedule or based on a trigger or event?
```

Comment:
I'm not sure what you mean either 😛 what aren't you sure about?

Reply:
@lwjohnst86 E.g., not sure what it means to "**conceptually** understand how a computer **might do a task** for someone"? :)

Reply:
Hmm, I do see the line you commented on, I am not sure what you aren't understanding or  asking. Maybe let's start with: What is it that you think it means?

Reply:
I think I know where you're going, but my confusion it twofold, I think:

- The wording "(conceptually)" and "might do" confuses me. It might be bc I know how it's done *practically* that I'm confused about the notion how to understand it **conceptually** only? As a reader I'm left with a feeling of "what do they **specifically** expect me to know here?"
- I'm also not sure why they are phrased as questions and not just statements like the rest of the list?

I think it means something in line with that the reader should understand something about that computers can do things for them (like renaming a file using the Terminal or a script, maybe, so they don't have to click on it an rename it?). And that it's not just when they "ask" it to do it (like running a command or a script), but also e.g., at specific times of day or at specific "events"? 

But it's not the hill I want to die on ⛰️  This PR has already been merged, so we can also just resolve this conversation.

Reply:
Thanks! That helps a lot more to understanding what you meant. And I wasn't going anywhere, I literally didn't understand what you were saying 😝 that's why I asked!

What if I wrote "conceptually or practically"? The conceptual part is important, because someone can conceptually *know* that computers can automate things, but not practically know how to implement it. Like we all conceptually know that we can structure data in a better way, but we don't fully know how that will be implemented. Some people don't even conceptually understand how computers work, so it would make no sense to talk to them about "automating things", when they will be like "what does that mean".

Good points about the question marks etc.

Reply:
All good! I get it. I also wasn't sure...

I have made a suggestion that makes more sense to me here: https://github.com/seedcase-project/seedcase-project/pull/238/files/bcc515fdd5ede60ba998e178d641197f2624c5c9#r1616295199